src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,348 @@
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+const renderVisualRefreshToolbarTriggerButton = (
+  props: Partial<VisualRefreshToolbarTriggerButtonProps> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VisualRefreshToolbarTriggerButton {...(renderProps as VisualRefreshToolbarTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(toolbarTriggerButtonStyles['trigger-wrapper'])!;
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+describe('Visual refresh trigger-button (not in appLayoutWidget toolbar)', () => {
+  beforeEach(() => jest.clearAllMocks());
+
+  describe.each([true, false])('Toolbar trigger-button with isMobile=%s', isMobile => {
+    describe.each([true, false])('AppLayoutInternals with hasOpenDrawer=%s', hasOpenDrawer => {
+      test('renders correctly without badge', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            badge: false,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(wrapper!.findIcon()).toBeTruthy();
+        expect(wrapper.findByClassName(visualRefreshStyles.dot)).toBeNull();
+        expect(wrapper.findBadge()).toBeNull();
+      });
+
+      test('renders correctly with aria controls adn aria label', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const mockAriaControls = 'mock-aria-control';
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton(
+          {
+            ariaControls: mockAriaControls,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(button!.getElement().getAttribute('aria-controls')).toEqual(mockAriaControls);
+      });
+
+      test('renders correctly with iconName', () => {
+        const ref: React.MutableRefObject<ButtonProps.Ref | null> = React.createRef();
+        const { wrapper, getByTestId } = renderVisualRefreshTriggerButton({}, {}, ref);
+
+        expect(wrapper).not.toBeNull();
```

Comment:
could it be combined with a previous test? the assertions look pretty much similar